libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -3440,20 +3448,20 @@ string YulUtilFunctions::conversionFunctionSpecial(Type const& _from, Type const
 			auto const& toArrayType = dynamic_cast<ArrayType const&>(_to);
 
 			solAssert(!fromArrayType.baseType()->isDynamicallyEncoded(), "");
-			solUnimplementedAssert(fromArrayType.isByteArray() && toArrayType.isByteArray(), "");
 			solUnimplementedAssert(toArrayType.location() == DataLocation::Memory, "");
 			solUnimplementedAssert(fromArrayType.location() == DataLocation::CallData, "");
 			solUnimplementedAssert(toArrayType.isDynamicallySized(), "");
 
 			Whiskers templ(R"(
 				function <functionName>(offset, length) -> converted {
 					converted := <allocateMemoryArray>(length)
-					<copyToMemory>(offset, add(converted, 0x20), length)
+					<copyToMemory>(offset, add(converted, 0x20), mul(length, <sizeInBytes>))
 				}
 			)");
 			templ("functionName", functionName);
 			templ("allocateMemoryArray", allocateMemoryArrayFunction(toArrayType));
 			templ("copyToMemory", copyToMemoryFunction(fromArrayType.location() == DataLocation::CallData));
+			templ("sizeInBytes", toArrayType.baseType()->memoryDataSize().str());
```

Comment:
And here dynamic...
Best I can do is to create one function that supports both copies and call from two different places.

I am afraid other options would require a lot of refactoring.
What do you think? @chriseth

Reply:
This does not look too complicated to me...

Reply:
Ah, now I see it. Does it make sense to extract an "array conversion function"? I don't think it would need much refactoring.

Reply:
That would make sense.
I can first do extracting in other PR, than revisit this one after moving array conversions to it's own function, ok?